File: 0704-binary-search/0704-binary-search.c
int search(int* nums, int numsSize, int target) {
    int low = 0;
    int high = numsSize - 1;
    int mid = -1;

    while (low <= high){
        mid  = (low + high) / 2;
        if (nums[mid] == target) return mid;
        else if (nums[mid] > target) high = mid - 1;
        else low = mid + 1;
    }

    return -1;
}

File: 0704-binary-search/0704-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    let low = 0
    let high = nums.length - 1

    let mid; 

    while (low <= high){
        mid = Math.floor ((high + low) / 2)
        if (nums[mid] === target) return mid
        if (nums[mid] < target) low = mid + 1
        if (nums[mid] > target) high = mid - 1
    }

    return -1
};